### DISCUSSION TITLE
Tuning Postal for a 6GB RAM VPS with 2 CPU Cores 30GB SSD to send over 100 emails/minute

### DISCUSSION THREAD

**Discussion by bukunmilab** [postalserver/postal#3164] [Help with using Postal] [answered]
Hey guys, I am playing around with the postal.yml and my.cnf to increase sending speed or tune postal to send faster on my **6GB RAM server with 2 CPU cores and 30 GB SSD**. I ran **htop** and I found that my server barely uses any resources even though I set 10 workers and 20 threads in **postal.yml** and 500 database connections in **etc/mysql/my.cnf file**.

Postal sending speed is way too slow but I believe I can tune a bunch of parameters to send faster. Right now here is what my sending speed looks like: 100 messages/ minute.

<img width="761" alt="Screenshot_2" src="https://github.com/user-attachments/assets/9a732f6f-0418-4345-8431-b3fe92959d2d">

Considering my server or VPS specs, what values would you suggest? My current values look like this:

/opt/postal/config/postal.yml

```
workers:
  quantity: 10
  threads: 20
```

/etc/mysql/my.cnf

```
[mysqld]
innodb_buffer_pool_size = 2048M
max_connections = 500
wait_timeout = 31536000
max_allowed_packet = 256M
innodb_flush_log_at_trx_commit = 2
```

**Comment by willpower232** [comment, marked as answer]
The worker quantity setting does not have any affect any more.

Presuming you're using the default installation, you can use the `--scale` argument to add more workers https://github.com/orgs/postalserver/discussions/1907#discussion-3939219

**Reply by bukunmilab** [reply]
Excellent! Problem solved, thank you.

**Reply by MichelHarryR** [reply]
your pbm is solved ? can you help ... when it put postal start --scale worker=20 ; what does it mean ?

**Reply by bukunmilab** [reply]
> your pbm is solved ? can you help ... when it put postal start --scale worker=20 ; what does it mean ?

You have to open your SSH terminal like Putty and enter the command based on what your server can handle. I have a 6GB RAM server with 2 CPU cores and 30GB SSD so I found 20workers as the most optimal I think my server can work with. Now it went from 1GB to 4.2GB RAM usage after I entered the command `postal start --scale worker=20` in SSH.

You may also want to first check your server resource usage with the command `htop` to determine if 20 is the right value for your use case. Play with different work values and see what your server can handle. CPU may not change that much but mine goes from 1.5% to 75%

**Reply by Areshkew** [reply]
What is your new pbm now?

**Comment by bukunmilab** [comment]
@willpower232 
By the way, since I run Postal in a docker using the automatic or default installation from Postal, there are too many docker containers. I mean it is difficult to find the actual my.cnf or mariadb.cnf file. I don't even know my database username/password or Mariadb username/password. And my server doesn't have this path: /etc/mysql/my.cnf NOR /etc/mysql/mariadb.cnf

Is it a good idea to create a global or optional file path in my server for this purpose like this:

/etc/mysql/my.cnf 
OR
/etc/mysql/mariadb.cnf

And I will configure with the parameters:

```
[mysqld]
innodb_buffer_pool_size = 2048M
max_connections = 1000
wait_timeout = 31536000
max_allowed_packet = 256M
innodb_flush_log_at_trx_commit = 2

```

In fact here is what I did. I copied one of those mariadb.cnf files from docker file path and use it as a template for the new path I created at /etc/mysql/my.cnf  

And I added the [mysqld] parameters defined above. Am I doing it right?
